tests/80torture/10filters.pl

```diff
@@ -0,0 +1,34 @@
+my $INVALID_FILTERS = [
+   { presence => "not_an_object" },
+   { room => { timeline => "not_an_object" } },
+   { room => { state => "not_an_object" } },
+   { room => { ephemeral => "not_an_object" } },
+   { room => { private_user_data => "not_an_object" } },
+   { room => { timeline => { "rooms" => "not_a_list" } } },
+   { room => { timeline => { "not_rooms" => "not_a_list" } } },
+   { room => { timeline => { "senders" => "not_a_list" } } },
+   { room => { timeline => { "not_senders" => "not_a_list" } } },
+   { room => { timeline => { "types" => "not_a_list" } } },
+   { room => { timeline => { "not_types" => "not_a_list"} } },
+   { room => { timeline => { "types" => [ 0 ] } } },
+   { room => { timeline => { "rooms" => [ "not_a_room_id" ] } } },
+   { room => { timeline => { "senders" => [ "not_a_sender_id" ] } } },
+];
+
+
+test "Check creating invalid filters returns 4xx",
+   requires => [qw( first_api_client )],
+
+   check => sub {
+      my ( $http ) = @_;
+
+      matrix_register_user( $http, undef, with_events => 0 )->then( sub {
```

Comment:
You can avoid some boilerplate and an entire nesting level here by doing

    requires => [ local_user_preparer() ],

    check => sub {
       my ( $user ) = @_;
       ...